packages/grid/_modules_/grid/hooks/features/selection/gridSelectionSelector.ts

```diff
@@ -3,15 +3,19 @@ import { gridRowsLookupSelector, GridRowsLookup } from '../rows/gridRowsSelector
 import { GridState } from '../core/gridState';
 import { GridSelectionState } from './gridSelectionState';
 import { GridRowId, GridRowModel } from '../../../models/gridRows';
+import { optionsSelector } from '../../utils/optionsSelector';
+import { GridOptions } from '../../../models/gridOptions';
 
 export const gridSelectionStateSelector = (state: GridState) => state.selection;
 export const selectedGridRowsCountSelector: OutputSelector<
   GridState,
   number,
-  (res: GridSelectionState) => number
-> = createSelector<GridState, GridSelectionState, number>(
+  (res: GridSelectionState, options: GridOptions) => number
+> = createSelector<GridState, GridSelectionState, GridOptions, number>(
   gridSelectionStateSelector,
-  (selection) => Object.keys(selection).length,
+  optionsSelector,
+  (selection, options) =>
+    options.selectionModel ? options.selectionModel.length : Object.keys(selection).length,
```

Comment:
you should not use the `options.selectionModel` here as the grid read the state. So the state is the source of truth and should reflect what is in options.

Reply:
Yes so if you set the model it updates the `state.selection`, then if you select a row, the state will be updated. Therefore, the source of truth is `state.selection`

Reply:
I can understand why we would want to keep the selector state only, we have never used the options before, it can feel scary, a new pattern. :+1: for trying a solution without.

For the prop updating the state or not. In https://github.com/mui-org/material-ui/blob/2c4dc994da164a53b253f03bcd2425a52655f175/packages/material-ui-utils/src/useControlled.js#L8 we skip the state. However, here, we have a double notion of prop controlled and state controlled, with the state with more priority. So we probably want to update the state.